docs/core/tools/custom-templates.md

```diff
@@ -0,0 +1,232 @@
+---
+title: Create custom templates for dotnet new (.NET Core SDK 2.0 Preview 2) | Microsoft Docs
+description: 
+keywords: dotnet new, CLI, CLI command, .NET Core, template, templating
+author: guardrex
+ms.author: mairaw
+ms.date: 06/26/2017
+ms.topic: article
+ms.prod: .net-core
+ms.technology: dotnet-cli
+ms.devlang: dotnet
+ms.assetid: c28eb59d-4d18-4aba-bcd1-88bd887abc08
+---
+
+# Create custom templates for dotnet new
+
+[!INCLUDE [core-preview-warning](~/includes/core-preview-warning.md)]
+
+The [.NET Core SDK](https://www.microsoft.com/net/download/core) provides many useful templates, but you can create your own custom templates for any type of .NET project, such as an app, service, tool, or class library. You can even create a template that outputs one or more independent files, such as a configuration file.
+
+Templates work with any Microsoft framework, including .NET Core, .NET Framework, and Xamarin. Users install your custom template from a NuGet package on any NuGet feed, by referencing a NuGet *nupkg* file directly, or by specifying a file system directory that contains the template. The template engine offers features that allow you to replace values, include and exclude files and regions of files, and execute custom processing operations when your template is used.
+
+The template engine is open source, and the online code repository is at [dotnet/templating](https://github.com/dotnet/templating/) on GitHub. You can open an issue if you run into a problem or start a discussion for a new feature, possibly a feature that you contribute. Visit the [dotnet/dotnet-template-samples](https://github.com/dotnet/dotnet-template-samples) repo for samples of templates. More templates, including templates from third parties, are found at [Available templates for dotnet new](https://github.com/dotnet/templating/wiki/Available-templates-for-dotnet-new) at GitHub. A detailed pre-release blog post that describes how to create and install custom templates written by [Sayed Ibrahim Hashimi](https://github.com/sayedihashimi) is at [How to create your own templates for dotnet new](https://blogs.msdn.microsoft.com/dotnet/2017/04/02/how-to-create-your-own-templates-for-dotnet-new/).
+
+## Configuration
+
+A template is composed of the following components:
+
+- Source files and folders
+- Configuration file (*template.json*)
+
+### Source files and folders
+
+The source files and folders are composed of whatever files and folders you want the template engine to use when the `dotnet new <TEMPLATE>` command is executed. The template engine is designed to use *runnable projects* as source code to produce projects. This has several benefits:
```

Comment:
> what if i want a non-runnable project as a source code like a config file or solution

Described below; however, I'm going to flesh it out a bit and explicitly call out "solution file" as a possible source/output for a template.